src/hooks/useFetch.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+
+import 'abortcontroller-polyfill';
+
+import { getAuthHeader, AUTH_ERROR_CODES } from 'sussol-utilities';
+
+import { SETTINGS_KEYS } from '../settings';
+import { UIDatabase } from '../database';
+
+const { SYNC_SITE_NAME, SYNC_SITE_PASSWORD_HASH } = SETTINGS_KEYS;
+const CONNECTION_TIMEOUT_PERIOD = 10000;
+
+const ERROR_CODES = {
+  ...AUTH_ERROR_CODES,
+  EMPTY_RESPONSE: 'No records found',
+};
+
+/* eslint-disable no-undef */
+const { AbortController } = window;
+
+const getAuthorizationHeader = () => {
+  const username = UIDatabase.getSetting(SYNC_SITE_NAME);
+  const password = UIDatabase.getSetting(SYNC_SITE_PASSWORD_HASH);
+  return { authorization: getAuthHeader(username, password) };
+};
+
+/**
+ * Simple custom hook to fetch data inside mounted component.
```

Comment:
I think because of the non-standardness(???) of this hook a little bit more of a description would be helpful?

`Custom hook which fetches data from the passed url ONLY as a component mounts. If the calling component is unmounted or the timeout is reached, the fetch is aborted. This should be used when data needs to be fetched as soon as a component mounts` or something .. you're way better explainerer then me

Reply:
5128a3a70b750e257bdd01398e38af80822b30bf.